libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1565,6 +1565,16 @@ void TypeChecker::endVisit(ElementaryTypeNameExpression const& _expr)
 
 void TypeChecker::endVisit(Literal const& _literal)
 {
+	if (_literal.looksLikeAddress())
+	{
+		if (_literal.passesAddressChecksum())
```

Comment:
The idea was that one class of typos is that you forget to type one character or type one character more than what would be correct. Because of that, `looksLikeAddress` is a bit sloppy and allows 39 to 41 characters, which then does not pass the checksum test and triggers a warning.